File: solutions/ruby/resistor-color-duo/4/resistor_color_duo.rb
=begin
Write your code for the 'Resistor Color Duo' exercise in this file. Make the tests in
`resistor_color_duo_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/resistor-color-duo` directory.
=end

class ResistorColorDuo
  RESISTOR_BAND = %w{black brown red orange yellow green blue violet grey white}
  def self.value(colors)
    colors
      .take(2)
      .map { |x| RESISTOR_BAND.index(x) }
      .join.to_i
  end
end

File: solutions/python/resistor-color-duo/2/resistor_color_duo.py
resistors_dict = {
    'black': 0,
    'brown': 1,
    'red': 2,
    'orange': 3,
    'yellow': 4,
    'green': 5,
    'blue': 6,
    'violet': 7,
    'grey': 8,
    'white': 9
}

def value(colors):
    band_values = [resistors_dict[colors[i].lower()] for i in range(2)]
    return int("".join(str(x) for x in band_values))